packages/network-controller/src/NetworkController.ts

```diff
@@ -435,6 +460,30 @@ export class NetworkController extends BaseControllerV2<
     }
     return isEIP1559Compatible;
   }
+
+  #setProviderAndBlockTracker({
```

Comment:
One of these would make sense to me:

1. provider and block tracker are always set together, never independently.
  - in this case, we want one method setting both together.
2. provider and block tracker are independent and can be set separately.
  - in this case, we want independent `setProvider` and `setBlockTracker`.

1 seems to be the case here. Given that this is a private method only called only with `provider.blockTracker` as `blockTracker` argument, why not simplify it into `#setProvider(provider)`?

Reply:
Oops, sorry, I didn't see this before.

This method is the opposite of `getProviderAndBlockTracker` — where with that method, you can get both, here you can set both. So, given that `getProviderAndBlockTracker` is a public method, I think it makes sense to keep this naming. Also, the NetworkController in the extension already has this pair of methods, so this PR simply ports them over and doesn't revise the API. That said, once we bring both the extension NetworkController and core NetworkController in alignment API-wise and once we've fully switched to the core NetworkController, I think we will have an opportunity to take a step back and discuss how we can improve the existing API. And maybe at that point, we can work out a simpler way to get the block tracker. For instance, given that the block tracker is used by the provider, maybe we can expose a property on the provider to access the block tracker rather than keep a separate reference to it. So then we really could have `setProvider` and `getProvider`.

In any case, would you mind if we tabled this discussion for later? I don't want to dismiss your suggestion, I'm just not sure if we can make those changes in this PR.